File: app/src/main/java/com/nikos_group/advancednotes/MainActivity.java
package com.nikos_group.advancednotes;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.nikos_group.advancednotes.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    public static AuthController authController = new AuthController();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(R.style.Theme_AdvancedNotes);
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        if (!authController.isAuth()) {
            startActivity(new Intent(this, LoginActivity.class),
                    ActivityOptions.makeCustomAnimation(this, R.anim.appear, R.anim.disappear).toBundle());
            finish();
        }

        new Thread(() -> {
            try {
                binding.getRoot().setAlpha(0f);
                Thread.sleep(500);
                runOnUiThread(() -> {
                    binding.getRoot().animate().alpha(1f).setDuration(250);
                });
            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
        }).start();

        binding.welcome.setOnClickListener(v -> {
            authController.signOut();
            startActivity(new Intent(this, LoginActivity.class),
                    ActivityOptions.makeCustomAnimation(this, R.anim.appear, R.anim.disappear).toBundle());
            finish();
        });
    }
}

File: app/src/main/java/com/nikos_group/advancednotes/activities/MainActivity.kt
package com.nikos_group.advancednotes.activities

import android.app.ActivityOptions
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import com.nikos_group.advancednotes.db.NoteDatabase
import com.nikos_group.advancednotes.repository.NoteRepository
import com.nikos_group.advancednotes.viewmodel.NoteViewModel
import com.nikos_group.advancednotes.viewmodel.NoteViewModelProviderFactory
import com.nikos_group.advancednotes.R
import com.nikos_group.advancednotes.authcontrollers.AuthController
import com.nikos_group.advancednotes.databinding.ActivityMainBinding
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    lateinit var noteViewModel: NoteViewModel

    companion object {
        val authController: AuthController = AuthController()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setUpViewModel()

        authentication()

        lifecycleScope.launch {
            transaction()
        }
    }

    private fun setUpViewModel() {
        val noteRepository = NoteRepository(
            NoteDatabase(this)
        )

        val viewModelProviderFactory = NoteViewModelProviderFactory(
            application,
            noteRepository
        )

        noteViewModel = ViewModelProvider(
            this,
            viewModelProviderFactory
        )[NoteViewModel::class.java]
    }

    private fun authentication() {
        if (!authController.isAuth) {
            startActivity(
                Intent(
                    this,
                    LoginActivity::class.java
                ),

                ActivityOptions.makeCustomAnimation(
                    this,
                    R.anim.appear,
                    R.anim.disappear
                ).toBundle()
            )

            finish()
        }
    }

    private suspend fun transaction() {
        binding.root.alpha = 0f
        delay(500)
        binding.root
            .animate()
            .alpha(1f)
            .duration = 250
    }
}